plugins/Marketplace/Marketplace.py

```diff
@@ -43,6 +41,7 @@ def show(self) -> None:
         If the window hadn't been loaded yet into Qt, it will be created lazily.
         """
         if self._window is None:
+            self._plugin_registry = PluginRegistry.getInstance()
```

Comment:
In my experience it makes it easier to test it, since you can use fixtures to prevent having to set the plugin registry every time

So instead of having to use context managers every time you make a call where you want the pluginregistry to be different, you can just use a fixture that creates a marketplace item with a different plugin registry.